types/lolex/index.d.ts

```diff
@@ -215,6 +284,22 @@ export interface LolexInstallOpts {
  *
  * @param now   Current time for the clock, as with lolex.createClock().
  * @param toFake   Names of methods that should be faked.
- * @type TClock   Type of clock to create.
+ * @type InstalledClock   Type of clock to create.
  */
-export declare function install<TClock extends Clock>(opts?: LolexInstallOpts): TClock;
```

Comment:
Instead of removing the `<TClock extends Clock>`, could you change it to `<TClock extends InstalledClock = InstalledClock>`? It used to be useful to specify which type of clock it was (browser vs node).

Reply:
Sure, will do. But could you explain to me what the = does? I didn't saw syntax like that in the documentation.

Reply:
These are [Generic parameter defaults](https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#generic-parameter-defaults). If the user doesn't provide an explicitly value for the generic, it'll default to whatever that value is. The docs are a little out of date on these newer things 🙁 but it's super nifty!

Come to think of it, maybe the `TClock` generic type should extend `Clock`, and the return type be `TClock & InstalledClock`? Something like _(but with better names)_:

```typescript
type InstalledClockBase = {
    uninstall: () => void;
    methods: FakeMethod[];
};

function install<TClock extends Clock>(opts?: LolexInstallOpts): TClock & InstalledClockBase;

type InstalledClock = InstalledClockBase & Clock;
```

```typescript
const browserClock = install<BrowserClock>(); // BrowserClock & InstalledClock;
const clock = install(); // InstalledClock
```